variable: unexpected failed test `TestSetTiDBCloudStorageURI`

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. Set a local [AWS credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html), say:
```
➜  .aws cat ~/.aws/credentials
[default]
aws_access_key_id=xxx
aws_secret_access_key=xxx

[bb7133]
aws_access_key_id=xxx
aws_secret_access_key=xxx
➜  .aws cat ~/.aws/config
[default]
region=xxx

[personal]
region=xxx
```

3. Run test [`TestSetTiDBCloudStorageURI`](https://github.com/pingcap/tidb/blob/ff8b3d356d604db05d8ca18e93d2c6b81db9c449/pkg/sessionctx/variable/sysvar_test.go#L1392)

### 2. What did you expect to see? (Required)

Success

### 3. What did you see instead (Required)

Failed with the following error message:

```
[FAIL]  pkg/sessionctx/variable TestSetTiDBCloudStorageURI
err=exit status 1
[2024/01/31 11:03:47.203 +08:00] [INFO] [s3.go:407] ["succeed to get bucket region from s3"] ["bucket region"=us-east-1]
--- FAIL: TestSetTiDBCloudStorageURI (1.80s)
    sysvar_test.go:1419:
        	Error Trace:	/Users/bb7133/Projects/gopath/src/[github.com/pingcap/tidb/pkg/sessionctx/variable/sysvar_test.go:1419](http://github.com/pingcap/tidb/pkg/sessionctx/variable/sysvar_test.go:1419)
        	Error:      	Error "check permission ListObjects failed due to AccessDenied: Access Denied\n\tstatus code: 403, request ...[BR:ExternalStorage:ErrStorageInvalidPermission]external storage permission" does not contain "bucket blackhole"
        	Test:       	TestSetTiDBCloudStorageURI
```

### 4. What is your TiDB version? (Required)

Latest master branch(ff8b3d356d604db05d8ca18e93d2c6b81db9c449)

--- a/pkg/sessionctx/variable/sysvar_test.go
+++ b/pkg/sessionctx/variable/sysvar_test.go
@@ -1416,7 +1416,7 @@ func TestSetTiDBCloudStorageURI(t *testing.T) {
 
 	// Set to s3, should fail
 	err = mock.SetGlobalSysVar(ctx, TiDBCloudStorageURI, "s3://blackhole")
-	require.ErrorContains(t, err, "bucket blackhole")
+	require.Error(t, err, "unreachable storage URI")
 
 	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(200)

